third-party/realdds/src/dds-participant.cpp

```diff
@@ -208,11 +206,14 @@ void dds_participant::init( dds_domain_id domain_id, std::string const & partici
     pqos.transport().use_builtin_transports = false;
     pqos.transport().user_transports.push_back( udp_transport );
 
+    _participant_factory = DomainParticipantFactory::get_shared_instance();
+
     // Listener will call DataReaderListener::on_data_available for a specific reader,
     // not SubscriberListener::on_data_on_readers for any reader
     // ( See note on https://fast-dds.docs.eprosima.com/en/v2.7.0/fastdds/dds_layer/core/entity/entity.html )
     StatusMask par_mask = StatusMask::all() >> StatusMask::data_on_readers();
-    _participant_factory = DomainParticipantFactory::get_shared_instance();
+    _domain_listener = std::make_shared< listener_impl >( *this );
```

Comment:
I don't quite understand how this works, can you clarify?
The listener is not in effect until we actually set it in the participant constructor, which is the last line in here. How's it getting called before then (or did I get the problem wrong)?

Reply:
Essentially you're saying that before the last line returns we already get a notification, so `_participant` isn't set.
I don't see how these changes would help...

Reply:
As discussed, I reverted the previous changes and added a check before the logging